core/src/main/java/org/everit/json/schema/ObjectSchema.java

```diff
@@ -418,4 +416,26 @@ public boolean definesProperty(String field) {
     return false;
   }
 
+  private boolean definesPatternProperty(String field) {
```

Comment:
You can rewrite this method to use the java8 stream API. It isn't a necessary change, but it could be good for practicing  it (if you are not familiar yet with streams), due to the simplicity of the task. The same goes for `definesSchemaDependencyProperty()`